### PR TITLE
NAT-301: allow EPiServer to find vacancies by location search

### PR DESCRIPTION
**Pull request by cnorthwood**
Initially this is postcode only, but as it's implemented as a helper function it'll be shared logic with Find Your Local Citizens Advice. There are a number of assumptions made at the moment that will need to be revisted when product decisions get fleshed out, e.g., number of results, should there be a max distance, etc.

### CODE REVIEW

File: app/controllers/api/v0/serialisers.rb [hunk 1]
```diff
@@ -63,6 +64,20 @@ def vacancy_as_v0_json(office)
...
+        vacancy = vacancy_as_v0_json(office)
+        vacancy[:distance] = distance_in_miles(location, office.location)
+        vacancy
+      end
+
+      def distance_in_miles(location1, location2)
+        if location1.nil? || location2.nil?
+          0
+        else
+          location1.distance(location2) / 1609.34
```
Comment (leeky) [inline app/controllers/api/v0/serialisers.rb:77]: Is it worth adding an explaining constant or a comment to explain that this number allows us to convert from metres into miles..?

File: spec/lss_loader_spec.rb [hunk 2]
```diff
@@ -61,7 +61,7 @@
 
     # Think the lat and lon look backwards? that's because in GIS coordinates are expressed in
     # x,y terms, not lat, lon. Therefore the lon comes first.
-    expect(Office.first.location).to eq RGeo::Cartesian.preferred_factory.point(1.18184, 51.07988)
+    expect(Office.first.location.as_text).to eq "POINT (1.18184 51.07988)"
```
Comment (leeky) [inline spec/lss_loader_spec.rb:64]: Why was this change necessary? I do, however, prefer static values to calculated ones, where they make sense.

**Review by leeky — APPROVED** [review]
I added a couple of minor comments, but otherwise this LGTM! :shipit:

File: app/controllers/api/v0/serialisers.rb (hunk 1 above)
Comment (cnorthwood) [reply app/controllers/api/v0/serialisers.rb:77]: I did wonder if pulling out `METRES_IN_A_MILE` constant would be needed for clarity, or if it'd be reasonably obvious from the name of the function it's in. Happy to switch it if needed!

File: spec/lss_loader_spec.rb (hunk 2 above)
Comment (cnorthwood) [reply spec/lss_loader_spec.rb:64]: well... so when I tried to use the distance algorithm in RGeo it was giving me the wrong answer, so when I googled it it's because I didn't let it know that this column actually corresponded to some geographical co-ordinates (by default it uses "geometry" mode, that is, unit-less x,y coordinates on a flat plane), so I added that in https://github.com/citizensadvice/local-office-search-api/pull/32/files#diff-aa4c01482b210e4d7cb2a4366b1dedca6bb7073488f1f08afd2da16d5fd6aa0cR4. Once that was in, this test failed because the Cartesian factory generates another unit-less point. I could have switched it to use a geography factory to build the right points, but I took this approach for simplicity
